### PR TITLE
Bug 1693260 - Add documentation about making a new release

### PR DESCRIPTION
**Pull request by brizental**

### CODE REVIEW

File: docs/release.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Making a new release
```
Comment (Dexterp37) [inline docs/release.md:1]: I think the SDK model works quite well for cutting a new release and it was ironed out through out the last year. We've been quite successful in applying the same model to Rally. Do you think there's value in doing the same for glean.js?

File: bin/prepare-release.sh [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    -e "s/\.\.\.main/...v${NEW_VERSION}/" \
+    "${WORKSPACE_ROOT}/${FILE}"
+run rm "${WORKSPACE_ROOT}/${FILE}.bak"
+
+if [ "$DOIT" = y ]; then
+    CHANGELOG=$(cat "${WORKSPACE_ROOT}/${FILE}")
+    cat > "${WORKSPACE_ROOT}/${FILE}" <<EOL
+# Unreleased changes
+
+[Full changelog](https://github.com/mozilla/glean/compare/v${NEW_VERSION}...main)
```
Comment (Dexterp37) [inline bin/prepare-release.sh:101]:
```suggestion
[Full changelog](https://github.com/mozilla/glean.js/compare/v${NEW_VERSION}...main)
```

**Review by Dexterp37 — APPROVED** [review]
r+wc

File: bin/prepare-release.sh (hunk 2 above)
Comment (brizental) [reply bin/prepare-release.sh:101]: Argh, I thought I got all of these. Thanks for catching it :D

**Pull request merged**